### PR TITLE
docs/adr: Incentivized testnet ADR

### PR DESCRIPTION
**Pull request by renaynay**

### CODE REVIEW

File: docs/adr/adr-007-incentivized-testnet.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+### Public API 
+The public API is currently being drafted [here](https://github.com/celestiaorg/celestia-node/pull/506/files). There are
+several components that will go into implementing the overall design as well as the individual module implementations.
+
+### [Bad Encoding Fraud Proof](https://github.com/celestiaorg/celestia-node/issues/528) (continuation)
+The full implementation of bad encoding fraud proof (BEFP) generation, propagation and handling is required for the 
+incentivized testnet. At the moment, generation and broadcasting of BEFPs is still in review. 
+
+### Testing TODO LINK TRACKING ISSUE
```
Comment (renaynay) [inline docs/adr/adr-007-incentivized-testnet.md:58]: @Bidon15 need you to fill in here

File: docs/adr/adr-007-incentivized-testnet.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+* Server-side request limits
+* Server-side maximum request size configuration
+* Adjacency of a batched `ExtendedHeader` response must be verified in `HeaderEx` upon receiving the response
+* Request retries
+* Cycling peers for requests
+
+More detail can be found in the issue linked in the header.
+
+### [IPLD v1 migration + NMT/IPLD optimization](https://github.com/celestiaorg/celestia-node/issues/614)
+@Wondertan todo
```
Comment (renaynay) [inline docs/adr/adr-007-incentivized-testnet.md:88]: @Wondertan any other issues you can link me to for this?

File: docs/adr/adr-007-incentivized-testnet.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+### [Telemetry](https://github.com/celestiaorg/celestia-node/issues/260)
+All node types must have metrics collection and tracing of hot paths implemented.
+
+## Nice to have
+### DB optimization
+@Wondertan TODO
```
Comment (renaynay) [inline docs/adr/adr-007-incentivized-testnet.md:95]: @Wondertan maybe dump some thoughts into a very basic issue ?

File: docs/adr/adr-007-incentivized-testnet.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+## Authors
+
+@renaynay @Wondertan
+
+## Context
+This ADR serves as a tracking document for celestia-node's technical requirements for the planned incentivized testnet.
+
+All epics/tracking issues have been linked in the sub-headers of [technical requirements](#technical-requirements).
+
+Top-level features will include a link to the epic-style tracking issue as well which can be used for a further detailed
```
Comment (adlerjohn) [inline docs/adr/adr-007-incentivized-testnet.md:16]: nit, but can we avoid hard line breaks in markdown files? I use side-by-side diffs and it looks atrocious at the particular width chosen (90??).

File: docs/adr/adr-007-incentivized-testnet.md (hunk 1 above)
Comment (Bidon15) [inline docs/adr/adr-007-incentivized-testnet.md:58]:
```suggestion
### [Testing](https://github.com/celestiaorg/celestia-node/issues/7)
```

File: docs/adr/adr-007-incentivized-testnet.md [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+incentivized testnet. At the moment, generation and broadcasting of BEFPs is still in review. 
+
+### Testing TODO LINK TRACKING ISSUE
+One of the biggest priorities for incentivized testnet is to test a lot of the assumptions we are working under in terms
+of sampling in realistic network conditions with a variety of topologies.
+
+Among the most important test cases, the following are required: 
+* TODO
+* TODO 
+* TODO
```
Comment (Bidon15) [inline docs/adr/adr-007-incentivized-testnet.md:65]:
```suggestion
* [Large Block Tests](https://github.com/celestiaorg/celestia-node/issues/602)
* [Block Recovery Tests](https://github.com/celestiaorg/test-infra/issues/21)
* [Peer Discovery Tests](https://github.com/celestiaorg/celestia-node/issues/649)
```

File: docs/adr/adr-007-incentivized-testnet.md [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+
+In terms of testing infrastructure, [testground](https://github.com/testground/testground) will be considered.
```
Comment (Bidon15) [inline docs/adr/adr-007-incentivized-testnet.md:67]:
```suggestion
In terms of testing infrastructure, [testground](https://github.com/testground/testground) will be considered. A [proof of concept](https://github.com/celestiaorg/celestia-node/issues/638) will be done in order to accomplish above tests
```

**Review by liamsi — COMMENTED** [review]
Besides the TODOs I think this looks fine.

File: docs/adr/adr-007-incentivized-testnet.md (hunk 3 above)
Comment (Wondertan) [reply docs/adr/adr-007-incentivized-testnet.md:95]: https://github.com/celestiaorg/celestia-node/issues/671

File: docs/adr/adr-007-incentivized-testnet.md [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+### [Telemetry](https://github.com/celestiaorg/celestia-node/issues/260)
+All node types must have metrics collection and tracing of hot paths implemented.
+
+## Nice to have
+### DB optimization
+@Wondertan TODO 
+* [Badgerv3 update](https://github.com/celestiaorg/celestia-node/issues/482)
+
+### [Share and header pruning](https://github.com/celestiaorg/celestia-node/issues/272)
+To make light nodes truly light, it would be nice to implement both share and `ExtendedHeader` pruning.
```
Comment (Wondertan) [inline docs/adr/adr-007-incentivized-testnet.md:99]: Let's make this a sub category of storage/Db optmizations

File: docs/adr/adr-007-incentivized-testnet.md [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+More detail can be found in the issue linked in the header.
+
+### [IPLD v1 migration + NMT/IPLD optimization](https://github.com/celestiaorg/celestia-node/issues/614)
+@Wondertan todo
+
+### [Telemetry](https://github.com/celestiaorg/celestia-node/issues/260)
+All node types must have metrics collection and tracing of hot paths implemented.
+
+## Nice to have
+### DB optimization
```
Comment (Wondertan) [inline docs/adr/adr-007-incentivized-testnet.md:94]: Let's call it disk usage/storage optimizations. I doubt we will do any DB engine performance optimizations(besides updating Badger)

File: docs/adr/adr-007-incentivized-testnet.md (hunk 2 above)
Comment (Wondertan) [reply docs/adr/adr-007-incentivized-testnet.md:88]: Let's remove IPLD v1 migration in favor of [this](https://github.com/celestiaorg/celestia-node/issues/656). NMT/IPLD storage optimizations can be kept.

File: docs/adr/adr-007-incentivized-testnet.md [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+
+All epics/tracking issues have been linked in the sub-headers of [technical requirements](#technical-requirements).
+
+Top-level features will include a link to the epic-style tracking issue as well which can be used for a further detailed
+breakdown of the individual feature requirements.
+
+## Legend
+
+**DA node**: any node type implemented in celestia-node
+**DA network**: the p2p network of celestia-node
```
Comment (adlerjohn) [inline docs/adr/adr-007-incentivized-testnet.md:23]:
```suggestion
- **DA node**: any node type implemented in celestia-node
- **DA network**: the p2p network of celestia-node
```

**Review by adlerjohn — APPROVED** [review]
fmt nits

**Review by liamsi — APPROVED** [review]

**Review by adlerjohn — APPROVED** [review]

**Review by Bidon15 — APPROVED** [review]

**Review by Wondertan — APPROVED** [review]

**Pull request merged**